src/jit/gentree.cpp

```diff
@@ -6399,27 +6399,11 @@ GenTree* Compiler::gtNewAssignNode(GenTree* dst, GenTree* src)
 // Return Value:
 //    Returns a node representing the struct value at the given address.
 //
-// Notes:
-//    It will currently return a GT_OBJ node for any struct type, but may
-//    return a GT_IND or a non-indirection for a scalar type.
-
-GenTree* Compiler::gtNewObjNode(CORINFO_CLASS_HANDLE structHnd, GenTree* addr)
+GenTreeObj* Compiler::gtNewObjNode(CORINFO_CLASS_HANDLE structHnd, GenTree* addr)
 {
     var_types nodeType = impNormStructType(structHnd);
     assert(varTypeIsStruct(nodeType));
 
-    if (!varTypeIsStruct(nodeType))
```

Comment:
Huh. I was thinking that this case would result in situations where we'd create an unnecessary `OBJ(ADDR)`, however, I now see that this could never actually be hit!